src/effects/effects_apply.c

```
@@ -64,6 +64,29 @@ static AttributeSet ReadAttributeSet
+	SchemaType t        // schema type (label / relationship-type)
+) {
+	int count = (t == SCHEMA_EDGE)
+		? Graph_RelationTypeCount (g)
+		: Graph_LabelTypeCount (g) ;
+
+	if (unlikely (id < 0 || id >= count)) {
+		// graph/effects-stream out of sync
+		RedisModule_Log (NULL, "warning",
+				"GRAPH.EFFECT refers to a missing %s (id: %d), graph is out of sync with the effects stream",
+				(t == SCHEMA_EDGE) ? "relationship-type" : "label", id) ;
+		exit (1) ;
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)